networkx/algorithms/tests/test_bridges.py

```diff
@@ -51,6 +53,53 @@ def test_multiedge_bridge(self):
         assert list(nx.bridges(G)) == [(2, 3)]
 
 
+class TestHasBridges:
+    """Unit tests for the has bridges function."""
+
+    def test_single_bridge(self):
+        edges = [
+            # DFS tree edges.
+            (1, 2),
+            (2, 3),
+            (3, 4),
+            (3, 5),
+            (5, 6),
+            (6, 7),
+            (7, 8),
+            (5, 9),
+            (9, 10),
+            # Nontree edges.
+            (1, 3),
+            (1, 4),
+            (2, 5),
+            (5, 10),
+            (6, 8),
+        ]
+        G = nx.Graph(edges)
+        source = 1
+        has_bridges = nx.has_bridges(G, source)
+        assert has_bridges == True
```

Comment:
This looks good! From a pure nit-picky standpoint it could be made slightly "leaner" by getting rid of some of the temporary variables. Since this test isn't really specific to the root argument, it might be nice to add multiple asserts: one for the default `root` and one where the user supplies it. Altogether, this chunk could be boiled down to something like:

```suggestion
        assert nx.has_bridges(G)  # Default root
        assert nx.has_bridges(G, root=1)  # arbitrary root in G
```